### PR TITLE
Feature: Refactor & move some of the components to func

### PR DESCRIPTION
**Pull request by VladyslavP**

### CODE REVIEW

File: src/components/Markets/index.tsx [hunk 1]
```diff
@@ -123,52 +62,71 @@ export class Markets extends React.Component<MarketsProps, MarketsState> {
...
+    }, [props.headers]);
 
-    private get filters() {
-        const { data } = this.props;
+    const createUniqueCurrencies = React.useCallback((currencies: string[], market: string) => {
+        const marketCurrencies = market.split('/').map((c: string) => c.trim());
+        const uniqueCurrencies = marketCurrencies.filter(c => !hasDuplicates(currencies, c));
+
+        return currencies.concat(uniqueCurrencies);
+    }, []);
```
Comment (oyershov) [inline src/components/Markets/index.tsx:84]: Seems this method can be replaced with ```currencies``` list from the redux

File: src/components/Markets/index.tsx [hunk 2]
```diff
@@ -123,52 +62,71 @@ export class Markets extends React.Component<MarketsProps, MarketsState> {
...
-    });
-}
+        ] : [];
+    }, [createUniqueCurrencies, filterType, transformCurrencyToFilter, data, filters]);
+
+    const getTableData = React.useCallback(() => {
+        const fd = data.filter(w => (w[0] as string).toLowerCase().includes(searchKey.toLowerCase()));
+
+        return fd.map(row => row.map(mapRows));
+    }, [data, mapRows, searchKey]);
```
Comment (oyershov) [inline src/components/Markets/index.tsx:112]: 1. Wrong name of method. Looks like here rather filter but not ```getTableData```
2. This looks to complex. Can you simplify code?

File: src/containers/OrderBook/index.tsx [hunk 3]
```diff
@@ -18,279 +16,219 @@ import {
...
-        const { isMobileDevice, forceLarge } = this.props;
+    const renderOrderBook = React.useCallback((array: string[][], side: string, message: string, currentM?: Market) => {
         let total = accumulateVolume(array);
-        const isLarge = forceLarge || this.state.width > breakpoint;
-        const priceFixed = currentMarket ? currentMarket.price_precision : 0;
-        const amountFixed = currentMarket ? currentMarket.amount_precision : 0;
+        const priceFixed = currentM ? currentM.price_precision : 0;
+        const amountFixed = currentM ? currentM.amount_precision : 0;
+        if (side === 'asks') {
+            total = isLarge ? total : accumulateVolume(array.slice(0).reverse()).slice(0).reverse();
```
Comment (oyershov) [inline src/containers/OrderBook/index.tsx:105]: Looks a bit weird
```suggestion
            total = isLarge ? total : accumulateVolume(array.slice(0).reverse()).slice(0).reverse();
```

**Comment by berezh** [comment]
Closed due to #372

**Pull request closed without merging**